sdlib/d/gc/tcache.d

```diff
@@ -421,3 +450,94 @@ unittest getCapacity {
 	assert(p5 !is p4);
 	assert(threadCache.getCapacity(p5[0 .. 16000]) == 20480);
 }
+
+unittest extend {
+	auto nonAppendable = threadCache.alloc(100, false);
+
+	// Attempt to extend a non-appendable:
+	assert(!threadCache.extend(nonAppendable[0 .. 100], 1));
+
+	// Make an appendable alloc:
+	auto p0 = threadCache.allocAppendable(100, false);
+	assert(threadCache.getCapacity(p0[0 .. 100]) == 16384);
+
+	// Attempt to extend slices with capacity 0:
+	assert(threadCache.extend(p0[0 .. 0], 0));
+	assert(!threadCache.extend(p0[0 .. 0], 50));
+	assert(!threadCache.extend(p0[0 .. 99], 50));
+	assert(!threadCache.extend(p0[1 .. 99], 50));
+	assert(!threadCache.extend(p0[0 .. 50], 50));
+
+	// Attempt extend with insufficient space:
+	assert(!threadCache.extend(p0[0 .. 100], 16285));
+	assert(!threadCache.extend(p0[50 .. 100], 16285));
```

Comment:
Extending by one less would prove that you tested the limit and that it is what you expect it to be.